rectools/model_selection/cross_validate.py

```diff
@@ -132,34 +100,26 @@ def cross_validate(  # pylint: disable=too-many-locals
             ]
         }
     """
-    interactions = dataset.interactions
-
-    split_iterator = splitter.split(interactions, collect_fold_stats=True)
+    split_iterator = splitter.split(dataset.interactions, collect_fold_stats=True)
 
     split_infos = []
     metrics_all = []
 
     for train_ids, test_ids, split_info in split_iterator:
         split_infos.append(split_info)
 
-        # ### Prepare split data
-        interactions_df_train = interactions.df.iloc[train_ids]  # 1x internal
-        # We need to avoid fitting models on sparse matrices with all zero rows/columns =>
-        # => we need to create a fold dataset which contains only hot users and items for current training
-        fold_dataset = _gen_2x_internal_ids_dataset(
-            interactions_df_train, dataset.user_features, dataset.item_features, prefer_warm_inference_over_cold
+        fold_dataset = dataset.filter_on_interactions_df_row_indexes(
+            row_indexes_to_keep=train_ids,
+            keep_external_ids=True,
+            keep_features_for_removed_entities=prefer_warm_inference_over_cold,
         )
+        interactions_df_test = dataset.interactions.df.loc[test_ids]
```

Comment:
Why don't you use your new dataset method here?

Reply:
But it'd be just one line of code instead of 3, no?

Reply:
It would be 2 lines here instead of 3.
5 lines in the func itself  and 20 lines for tests :)
right now the logic is clear and all is fine. no need for dataset when we need  just interactions

Reply:
Didn't get your point, sorry. 
You have the func already, just call it and take interactions.  
It'd be more readable than doing manual manipulations.  
Why do you need add lines to the func?

Reply:
It's not appropriate since new func is for dataset creation. It has more going on than just converting ids of interactions to external ones. For example it will construct new ids maps. We just don't need it here